submission/persistence-filesystem/src/main/kotlin/ac/uk/ebi/biostd/persistence/filesystem/service/FileProcessingService.kt

```diff
@@ -0,0 +1,60 @@
+package ac.uk.ebi.biostd.persistence.filesystem.service
+
+import ac.uk.ebi.biostd.persistence.filesystem.fire.fireCopy
+import ac.uk.ebi.biostd.persistence.filesystem.fire.fireMove
+import ac.uk.ebi.biostd.persistence.filesystem.nfs.nfsCopy
+import ac.uk.ebi.biostd.persistence.filesystem.nfs.nfsMove
+import ac.uk.ebi.biostd.persistence.filesystem.request.FileProcessingRequest
+import arrow.core.Either
+import ebi.ac.uk.extended.model.ExtFile
+import ebi.ac.uk.extended.model.ExtFileList
+import ebi.ac.uk.extended.model.ExtFileTable
+import ebi.ac.uk.extended.model.ExtSection
+import ebi.ac.uk.extended.model.ExtSectionTable
+import ebi.ac.uk.extended.model.ExtSubmission
+import ebi.ac.uk.extended.model.FileMode.COPY
+import ebi.ac.uk.extended.model.FireFile
+import ebi.ac.uk.extended.model.NfsFile
+
+class FileProcessingService {
+    fun processFiles(processRequest: FileProcessingRequest): ExtSubmission {
+        val (mode, submission, config) = processRequest
+
+        fun processFile(file: ExtFile): ExtFile = when (file) {
```

Comment:
not clear for me why we are handling both scenarios here, this class is used by `NfsFilesService`  which only receive NFs request. I think we should not share `FileProcessingService`  for both `FilesService` implementation

Reply:
Nice catch. Fixed